[Game logic]

BeaverWIthKnife opened the pull request:
2 TODO done:
color in boats and grids are now private and have getters and setters
Added:
GetIsGameOver();
CheckIsGameOver(); (if all ships are dead for one of the players)

Edited:
IsAllShipsPlaced() from color check to bool check
Game::update now updates if game is over

Refactoring

BeaverWIthKnife committed:
[Game logic]
2 TODO done:
color in boats and grids are now private and have getters and setters
Added:
GetIsGameOver();
CheckIsGameOver(); (if all ships are dead for one of the players)

Edited:
IsAllShipsPlaced() from color check to bool check
Game::update now updates if game is over

BeaverWIthKnife committed:
[Refactoring]
GetIsDead -> IsDead
const bool &boat -> const bool isBoat

roma4004 commented on src/Game.cpp:
```diff
@@ -307,7 +307,7 @@ bool Game::MakeShot(const SDL_Rect &shotRect, const SDL_Rect &damageRect, vector
     if (auto damagedBoat = ShotCollideOtherBoats(damageRect, playerBoats); damagedBoat != nullptr) {
         if (IsAllShipPartDamaged(damagedBoat)) {
-            damagedBoat->SetIsDead(true);
+            damagedBoat->SetIsDead(TODO);
```
Try to compile code base before pushing any commit

BeaverWIthKnife committed:
[Fix]
forgot TODO line 310 Game.cpp

BeaverWIthKnife committed:
[Refactoring]
Cleanup + renaming

BeaverWIthKnife committed:
[Formatting]

BeaverWIthKnife committed:
[Game logic refactoring]
Check if game is over moved from update to shot

roma4004 commented on src/Game.cpp:
```diff
@@ -14,13 +14,22 @@ Game::~Game() = default;
+}
+
+void Game::update() {
+    _gameOver = CheckIsGameOver();
```
Calling frequency is too high, we can do it once we hit or even better when we kill ships. 

This significantly reduces the number of operations to detect end-of-game status.

So method update isn't needed any more, for now.

BeaverWIthKnife committed:
Revert "[Game logic refactoring]"

This reverts commit 07347e27d8138f9ed13a0043cf22043d760aef09.

BeaverWIthKnife committed:
[Game logic refactoring]
Check if game is over moved from update to moment where ship is killed
Change of all ships killed logic -> now counted instead of checking if every boat is dead
In case of not ++ counter everytime ship is added - we assign counter when the game state is changed
light refactoring (adding const where is needed)

roma4004 committed:
renaming and code cleanup
